lib/Launcher.js

```diff
@@ -190,6 +191,31 @@ class Launcher {
   }
 }
 
+class ExitEmitter extends EventEmitter {
+  constructor() {
+    super();
+    this.setMaxListeners(Infinity);
+    const processListeners = [ helper.addEventListener(process, 'exit', exitHandler.bind(this)) ];
```

Comment:
Essentially the single array that will hold all the SIG and exit listeners that are added to the process. This class is only initialized once at the module level so process will only get these added once.